token.go

```diff
@@ -77,19 +77,19 @@ func (t *tokenMan) maintainToken() {
 
 func (t *tokenMan) invalidateToken() {
 	log.Debug("invalidating token")
-	getTokenLock.Lock()
+	t.Lock()
 	t.token = nil
+	t.Unlock()
```

Comment:
moving the unlock here allows a call to getToken() to return a nil token, as we talked about.  Let's see if we can restore the unlock after the refresh confirmation is received.  Hopefully this doesn't introduce new races.  If it does, we'll go from there.